vernac/synterp.ml

```diff
@@ -282,7 +282,22 @@ let _ = CErrors.register_handler begin function
   | _ -> None
 end
 
+let warn_deprecated_from_Coq =
+  CWarnings.create ~name:"deprecated-from-Coq"
+    ~category:CWarnings.CoreCategories.deprecated
+    (fun () -> strbrk
+        "\"From Coq\" has been replaced by \"From Stdlib\".")
+
+let deprecated_Coq p =
+  let p', i = Libnames.repr_qualid p in
+  if not (DirPath.is_empty p')
+     || not (String.equal (Id.to_string i) "Coq") then p
```

Comment:
I'd rather do
```suggestion
  if not (Libnames.qualid_eq p (Libnames.qualid_of_string "Coq")) then p
```

Reply:
also I guess this won't work if someone does `From Coq.Classes Require Morphisms`, do we care?

Reply:
No, this is handled in nametab.ml, so we only need to handle `From Coq Require` here.